subsys/bluetooth/mesh/net.c

```diff
@@ -661,18 +662,17 @@ static void bt_mesh_net_relay(struct net_buf_simple *sbuf,
 		goto done;
 	}
 
-	/* Sending to the GATT bearer should only happen if GATT Proxy
-	 * is enabled.
+	/* When the Friend node relays message for lpn, the message will be
+	 * retransmitted using the managed master security credentials and
+	 * the Network PDU shall be retransmitted to all network interfaces.
 	 */
 	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
-	    bt_mesh_gatt_proxy_get() == BT_MESH_GATT_PROXY_ENABLED) {
-		if (bt_mesh_proxy_relay(&buf->b, rx->ctx.recv_dst) &&
-		    BT_MESH_ADDR_IS_UNICAST(rx->ctx.recv_dst)) {
-			goto done;
-		}
+	    (rx->friend_cred ||
+	     bt_mesh_gatt_proxy_get() == BT_MESH_GATT_PROXY_ENABLED)) {
+		bt_mesh_proxy_relay(&buf->b, rx->ctx.recv_dst);
```

Comment:
Unless it breaks qualification, I think the old unicast address check should still be here. If a unicast message is going to a proxy node, there's no reason to relay it to the advertising bearer.

Reply:
Yes, it is suitable for white list, but if it is for black list, even unicast address does not mean it is required by proxy client.

Reply:
I am thinking, if blacklist filtering is enabled, if the unicast address is not in the blacklist, does it mean that the proxy client needs this unicast address? In this case, it does not need to be relayed through ADV bearer.

Reply:
Good point, I forgot the proxy puts the unicast addresses in the filters. I suppose we can only be sure of this check if the proxy is using a whitelist. Your original version was better, I didn't consider the blacklist.